indigo/analysis/refs/works.py

```diff
@@ -0,0 +1,184 @@
+import re
+
+from django.db.models import Q
+
+from cobalt import FrbrUri
+from docpipe.matchers import CitationMatcher
+from indigo.analysis.matchers import DocumentPatternMatcherMixin
+from indigo.plugins import plugins
+from indigo_api.models import Country, Work, WorkAlias
+from indigo_api.models.citations import CitationAlias
+
+
+class CommonTitlesCitationMatcher(DocumentPatternMatcherMixin, CitationMatcher):
+    """Finds references to works based "the XXX Act" patterns. """
+    # only supports English
+    locale = (None, 'eng', None)
+
+    html_candidate_xpath = ".//text()[contains(., ' the ') and contains(., ' Act') and not(ancestor::a)]"
+    xml_candidate_path = ".//text()[contains(., ' the ') and contains(., ' Act') and not(ancestor::ns:ref)]"
+
+    pattern_re = re.compile(r'\b(?P<pre>the\s+(?P<title>((?!the)\S+\s+){1,7})Act)\s*,?(\s*of\s+)?\s*(?P<year>[12][0-9]{3})?\b')
```

Comment:
am I reading this right that "XXX" in 'the XXX Act' can only be 7 characters long and contain no spaces?

Reply:
no, it can be up to 7 words long `(\S+\s+){1,7}`

Reply:
the makes a lot more sense :)